### DISCUSSION TITLE
M1 Spins uncontrollably when connected to power, but CF2 not turned on

### DISCUSSION THREAD

**Discussion by lbult** [bitcraze/discussions#726] [Q&A]
Hi!

I have an issue where M1 spins at (near) full throttle when the battery is connected to the Crazyflie, but the Crayzflie itself is not turned on. This also happens when the Crazyflie is in bootloader mode. Some things that I tried: flash the firmware, disconnect the deck that is attached and change the motor.

Do you have any idea how to repair the Crazyflie or fix this issue? Let me know if there are any additional things that I could try or test.

**Comment by krichardsson** [comment]
Hi!

It sounds like something is broken in the motor driver. You can find the design in the first drawing in [the schematic](https://www.bitcraze.io/documentation/hardware/crazyflie_2_1/crazyflie_2.1_schematics_rev.b.pdf).

Tobias, our hardware expert is enjoying a week off, he can probably help you next week.

**Comment by tobbeanton** [comment]
Most likely the M1 mosfet is broken. It is the 3-legged SOT-23 chip on the bottom side, close to the M1 arm, has the marking S2 and is close to PA3 PCB marking. If you have some basic soldering skill you can replace it with e.g a [PMV30UN2](http://www.findchips.com/search/PMV30UN2). Easiest way to remove the old chip is to cut the legs before removing the remaining part with the solder iron.

**Comment by MattEll1** [comment]
Hello everyone, I am having a very similar issue with the CrazyFlie 2.1. When powered off with the battery connected, the top right motor spins continuously. However after powering it on and completing the self-test spin sequence,  it functions as usual and can fly. This did not always happen. Could this be the same MOSFET issue? 

I do not know if this is related, but for context, I have recently been testing out PX4 on the CrazyFlie 2.1. So I have been using [the guide](https://docs.px4.io/main/en/complete_vehicles_mc/crazyflie21) to swap between the CrazyFlie and PX4 bootloader and firmware. I did this maybe 3 or 4 times, so not a whole lot. But, this issue started to occur when I swapped from PX4 back to the CrazyFlie firmware the last time.

I've tried re-flashing the CrazyFlie bootloader and latest firmware, and I've tried doing a cold reboot. Neither solved the issue.  I don't believe I did anything that could damage the MOSFET between running PX4 and flashing back to the Crazyflie firmware. Also, if the MOSFET was damaged, would the motor still function as normal once powered on? 

I am still very new to all of this, so I'm not sure if I am having the same issue. I would really appreciate any guidance on this. I've shared a video below demonstrating:

https://github.com/user-attachments/assets/b24be319-e788-4ef6-8939-4a6a22b1ee64

**Reply by tobbeanton** [reply]
My guess is still that something is strange with the mosfet but it could also be the pull-down resistor that is broken for some reason. You can measure it with a multimeter
![image](https://github.com/user-attachments/assets/5da63e50-2a73-4e48-82c4-7e82d1e864cb)
The resistor marked in blue should read ~47ohm and yellow 10k

**Reply by MattEll1** [reply]
Hi, apologies for the late response.
For the blue resistor I'm getting 47.2 ohms. However for the yellow resistor I'm getting 8k ohms. Do you think that could be the issue?

**Reply by tobbeanton** [reply]
Then I think the MOSFET is broken, it is the SOT23 package just by the resistors.
